ctags/parsers/asciidoc.c

```diff
@@ -78,7 +78,7 @@ static void makeAsciidocTag (const vString* const name, const int kind)
 	{
 		tagEntryInfo e;
 
-		initTagEntry (&e, vStringValue (name), &(AsciidocKinds [kind]));
+		initTagEntry (&e, vStringValue (name), kind);
 
 		e.lineNumber--;	/* we want the line before the '---' underline chars */
 
```

Comment:
I suspect the scope information that used to be stored in extensions about here is the cause of the breakage of asciidoc structuring and similar change for rest.  Needs to be fixed, note that if upstream won't accept it then these two parsers will have to remain Geany specific.

Reply:
No, this is correct - it's the "Use kindIndex instead of kindDefinition" patch mentioned above. It's now enough to use the index (which is "kind" here) instead of the value from the structure.

But the problem seems to be this patch:

https://github.com/geany/geany/commit/3e8365c7f2285c32ef1372f45afd698449a01ac4#diff-349c047a4c33ab772a7ad84be2c13b4c

Upstream there's some additional comment why using cork isn't good in this case:

https://github.com/universal-ctags/ctags/blob/master/parsers/asciidoc.c

I'll prepare a patch tomorrow to fix that.

Reply:
yes, the lines that patch was removing was what I was referring to.